File: .idea/app/src/main/java/com/chevstrap/rbx/UI/Elements/Settings/Pages/Chevstrap.java
package com.chevstrap.rbx.UI.Elements.Settings.Pages;

import android.content.Context;
import android.widget.LinearLayout;

import com.chevstrap.rbx.App;
import com.chevstrap.rbx.Enums.RobloxAppType;
import com.chevstrap.rbx.Enums.ThemeRecreated;
import com.chevstrap.rbx.R;
import com.chevstrap.rbx.UI.ViewModels.Settings.BehaviourViewModel;
import com.chevstrap.rbx.UI.ViewModels.Settings.ChevstrapViewModel;
import com.chevstrap.rbx.Models.MethodPair;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.lang.reflect.Method;

public class Chevstrap {

    public static void addEveryPresets(Context context, LinearLayout parentLayout, ChevstrapFragment fragment) {
        fragment.addSection(parentLayout, getTextLocale(context, R.string.common_general));

        fragment.addAccordionMenu(getTextLocale(context, R.string.menu_settings_accordion_title_appearence), getTextLocale(context, R.string.menu_settings_accordion_description_appearence), () -> addA(context, fragment), 1);
//        try {
//            JSONArray array = new JSONArray();
//
//            for (LanguageRecreateds mode : LanguageRecreateds.values()) {
//                JSONObject obj = new JSONObject()
//                        .put("label", mode.getDisplayName())
//                        .put("value", mode.name());
//                array.put(obj);
//            }
//
//            Method getMethod = SettingsViewModel.class.getMethod("getAppLanguage");
//            Method setMethod = SettingsViewModel.class.getMethod("setAppLanguage", String.class);
//
//            MethodPair methods = new MethodPair(getMethod, setMethod);
//            fragment.addDropdown(getTextLocale(context, R.string.menu_settings_language_title), getTextLocale(context, R.string.menu_settings_language_description), parentLayout, array, methods);
//        } catch (JSONException ignored) {
//        } catch (NoSuchMethodException e) {
//            throw new RuntimeException(e);
//        }
    }

    public static void addA(Context context, ChevstrapFragment fragment) {
        try {
            JSONArray array = new JSONArray();

            for (ThemeRecreated mode : ThemeRecreated.values()) {
                JSONObject obj = new JSONObject()
                        .put("label", mode.getDisplayName())
                        .put("value", mode.name());
                array.put(obj);
            }

            Method getMethod = ChevstrapViewModel.class.getMethod("getapp_theme_in_app");
            Method setMethod = ChevstrapViewModel.class.getMethod("setapp_theme_in_app", String.class);

            MethodPair methods = new MethodPair(getMethod, setMethod);
            fragment.addDropdown(getTextLocale(context, R.string.menu_settings_color_theme_in_app_title), getTextLocale(context, R.string.menu_settings_color_theme_in_app_description), array, methods, 1);
        } catch (JSONException e) {
//            e.printStackTrace();
        } catch (NoSuchMethodException e) {
            throw new RuntimeException(e);
        }
        fragment.addImportButton(getTextLocale(App.getAppContext(), R.string.menu_settings_background_video_title), getTextLocale(App.getAppContext(), R.string.menu_settings_background_video_description), null, "set_animated_theme",  1);

        try {
            Method getMethod = ChevstrapViewModel.class.getMethod("isForcePotraitMode");
            Method setMethod = ChevstrapViewModel.class.getMethod("setForcePotraitMode", boolean.class);
            MethodPair methods = new MethodPair(getMethod, setMethod);
            fragment.addToggle(
                    getTextLocale(context, R.string.menu_settings_force_portrait_mode_title),
                    getTextLocale(context, R.string.menu_settings_force_portrait_mode_description),
                    methods, 1
            );
        } catch (NoSuchMethodException e) {
            throw new RuntimeException(e);
        }
    }

    private static String getTextLocale(Context context, int resId) {
        return context.getString(resId);
    }
}

File: app/src/main/java/com/chevstrap/rbx/ui/views/settings/pages/Chevstrap.kt
package com.chevstrap.rbx.ui.views.settings.pages

import android.content.Context
import android.widget.LinearLayout
import com.chevstrap.rbx.App
import com.chevstrap.rbx.R
import com.chevstrap.rbx.enums.ThemeRecreated
import com.chevstrap.rbx.extensions.ResourceManagerEx
import com.chevstrap.rbx.ui.viewModels.MethodAction
import com.chevstrap.rbx.ui.viewModels.MethodPair
import com.chevstrap.rbx.ui.viewModels.settings.pages.ChevstrapViewModel
import org.json.JSONArray
import org.json.JSONObject

object Chevstrap {

    private val viewModel = ChevstrapViewModel()

    fun addEveryPresets(
        context: Context,
        parentLayout: LinearLayout,
        fragment: ChevstrapFragment
    ) {

        fragment.addSection(
            parentLayout,
            ResourceManagerEx.getStringOrEmpty(
                context,
                R.string.common_general
            )
        )

        fragment.addAccordionMenu(
            ResourceManagerEx.getStringOrEmpty(
                context,
                R.string.menu_settings_accordion_title_appearence
            ),
            ResourceManagerEx.getStringOrEmpty(
                context,
                R.string.menu_settings_accordion_description_appearence
            ),
            {
                addA(
                    context,
                    fragment
                )
            },
            1
        )

        fragment.addButton(
            ResourceManagerEx.getStringOrEmpty(
                App.appContext!!,
                R.string.menu_settings_open_logs_viewer_title
            ),
            ResourceManagerEx.getStringOrEmpty(
                App.appContext!!,
                R.string.menu_settings_open_logs_viewer_description
            ),
            MethodAction { viewModel.logExplorer() },
            -1
        )
    }

    private fun addA(
        context: Context,
        fragment: ChevstrapFragment
    ) {

        val array = JSONArray()
        for (mode in ThemeRecreated.entries) {
            array.put(
                JSONObject()
                    .put(
                        "label",
                        mode.displayName
                    )
                    .put(
                        "value",
                        mode.name
                    )
            )
        }

        fragment.addDropdown(
            ResourceManagerEx.getStringOrEmpty(
                context,
                R.string.menu_settings_app_theme_in_app_title
            ),
            "",
            array,
            MethodPair(
                viewModel,
                ChevstrapViewModel::appThemeInApp
            ),
            1
        )

        fragment.addButton(
            ResourceManagerEx.getStringOrEmpty(
                App.appContext!!,
                R.string.menu_settings_background_image_title
            ),
            ResourceManagerEx.getStringOrEmpty(
                App.appContext!!,
                R.string.menu_settings_background_image_description
            ),
            MethodAction { viewModel.importCustomAppBackground() },
            1
        )
    }
}